net/golioth/system_client.c

```diff
@@ -159,12 +147,38 @@ static int golioth_check_credentials(const uint8_t *psk_id, size_t psk_id_len,
 	return err;
 }
 
+static int golioth_check_cert_credentials(void)
+{
+	size_t cred_len = 0;
+	int err = tls_credential_get(CONFIG_GOLIOTH_SYSTEM_CLIENT_CREDENTIALS_TAG,
+				     TLS_CREDENTIAL_SERVER_CERTIFICATE, NULL, &cred_len);
+	if (-ENOENT == err) {
```

Comment:
```suggestion
	if (err == -ENOENT) {
```
And how about other error codes?

Reply:
This was put in specifically to guard against https://github.com/pion/dtls/issues/529 (spent about 30 minutes confused before I remembered that one). I don't think there's value in checking other error codes. Right now, the only other one that is specified in the API is `-EACCESS`. This isn't actually returned anywhere in the implementation, but I imagine it's there in the case that a future TLS Credentials subsystem implementation uses a secure storage backend, which could allow writing keys but not reading them, or only allow reading keys from secure code.